src/main/java/net/countercraft/movecraft/combat/features/directors/AADirectors.java

```diff
@@ -95,44 +96,43 @@ private void processFireball(@NotNull SmallFireball fireball) {
         fireball.setShooter(p);
 
         Vector fireballVector = fireball.getVelocity();
-        double speed = fireballVector.length(); // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
+        double speed = fireballVector.length() * FireballSpeed; // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
         fireballVector = fireballVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
 
         Block targetBlock = DirectorUtils.getDirectorBlock(p, AADirectorRange);
         Vector targetVector;
-        if (targetBlock == null || targetBlock.getType().equals(Material.AIR)) // the player is looking at nothing, shoot in that general direction
+
+        if (targetBlock == null || targetBlock.getType().isAir()) // the player is looking at nothing, shoot in that general direction
             targetVector = p.getLocation().getDirection();
         else { // shoot directly at the block the player is looking at (IE: with convergence)
             targetVector = targetBlock.getLocation().toVector().subtract(fireball.getLocation().toVector());
             targetVector = targetVector.normalize();
         }
 
-        if (targetVector.getX() - fireballVector.getX() > 0.5)
-            fireballVector.setX(fireballVector.getX() + 0.5);
-        else if (targetVector.getX() - fireballVector.getX() < -0.5)
-            fireballVector.setX(fireballVector.getX() - 0.5);
-        else
-            fireballVector.setX(targetVector.getX());
-
-        if (targetVector.getY() - fireballVector.getY() > 0.5)
-            fireballVector.setY(fireballVector.getY() + 0.5);
-        else if (targetVector.getY() - fireballVector.getY() < -0.5)
-            fireballVector.setY(fireballVector.getY() - 0.5);
-        else
-            fireballVector.setY(targetVector.getY());
-
-        if (targetVector.getZ() - fireballVector.getZ() > 0.5)
-            fireballVector.setZ(fireballVector.getZ() + 0.5);
-        else if (targetVector.getZ() - fireballVector.getZ() < -0.5)
-            fireballVector.setZ(fireballVector.getZ() - 0.5);
-        else
-            fireballVector.setZ(targetVector.getZ());
+        double x = fireballVector.getX();
+        fireballVector.setX( x + adjustValue(targetVector.getX() - x) );
+
+        double y = fireballVector.getY();
+        fireballVector.setY( y + adjustValue(targetVector.getY() - y) );
+
+        double z = fireballVector.getZ();
+        fireballVector.setZ( z + adjustValue(targetVector.getZ() - z) );
 
         fireballVector = fireballVector.multiply(speed); // put the original speed back in, but now along a different trajectory
         fireball.setVelocity(fireballVector);
         fireball.setDirection(fireballVector);
     }
 
+    private double adjustValue(double value) {
```

Comment:
Let's rewrite this to take the `actual` and `target` like so:
```java
if (target - actual > 0.5)
       return actual + 0.5;
else if (target - actual < -0.5)
       return actual - 0.5;

return target;
```

It will greatly simplify the above, which can become like so:
```java
fireballVector.setX(adjustValue(targetVector.getX(), fireballVector.getX()));
```